Services/Init/classes/class.ilPasswordAssistanceGUI.php

```diff
@@ -115,55 +110,83 @@ public function executeCommand()
         }
     }
 
-    /**
-     * Returns the ILIAS http path without a trailing /
-     */
-    protected function getBaseUrl(): string
+    private function getBaseUrl(): string
     {
-        return rtrim(ILIAS_HTTP_PATH, '/');
+        return rtrim(ilUtil::_getHttpPath(), '/');
     }
 
     /**
-     * @param string $script
-     * @param array  $queryParameters
-     * @return string
+     * @param array<string, string> $query_parameters
      */
-    protected function buildUrl(string $script, array $queryParameters): string
+    private function buildUrl(string $script, array $query_parameters): string
     {
         $url = implode('/', [
             $this->getBaseUrl(),
             ltrim($script, '/')
         ]);
 
-        $url = \ilUtil::appendUrlParameterString(
+        $url = ilUtil::appendUrlParameterString(
             $url,
-            http_build_query($queryParameters, null, '&')
+            http_build_query($query_parameters, '', '&')
         );
 
         return $url;
     }
 
-    protected function getAssistanceForm(): ilPropertyFormGUI
+    private function emailTrafo(): \ILIAS\Refinery\Transformation
     {
-        $form = new ilPropertyFormGUI();
-
-        $form->setTitle($this->lng->txt('password_assistance'));
-        $form->setFormAction($this->ctrl->getFormAction($this, 'submitAssistanceForm'));
-        $form->setTarget('_parent');
+        return $this->refinery->custom()->constraint(
+            static function ($value): bool {
+                return is_string($value) && ilUtil::is_email($value);
+            },
+            $this->lng->txt('email_not_valid')
+        );
+    }
```

Comment:
I would suggest adding an email-transformation to the refinery as this looks like a very common kind of data to be validated

Reply:
Or even better: An `Email` data type for `.src/Data`?

Reply:
@klees What do you think? `Transformation` oder `Data Type`? None of these suggestions?

If we add an alternative to `ilUtil::is_email`, we should re-use (or move code from) `ilMailRfc822AddressParserFactory`.

Reply:
@pascalseeland IMO we should merge this and add a TODO for the ROADMAP